### PR TITLE
Increased Coverage for Slack Management files

### PR DESCRIPTION
**Pull request by Yashgupta9330**
Resolves https://github.com/OWASP/Nest/issues/193

 I have added test files for the slack/management/commands and commands and events files, which have increased the test coverage to 52%.

image
image

### CODE REVIEW

File: backend/apps/common/utils.py [hunk 1]
```diff
@@ -34,7 +34,11 @@ def join_values(fields, delimiter=" "):
 
 def natural_date(value):
     """Return humanized version of a date."""
-    return naturaltime(datetime.fromtimestamp(value, tz=timezone.utc))
+    if isinstance(value, str):
+        value = datetime.strptime(value, "%Y-%m-%d").replace(tzinfo=timezone.utc)
+    elif isinstance(value, int):
+        value = datetime.fromtimestamp(value, tz=timezone.utc)
+    return naturaltime(value)
```
Comment (arkid15r) [inline backend/apps/common/utils.py:41]: Could you add tests for this update too?

File: backend/tests/slack/commands/projects_tests.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+def test_handler_with_results(
+    mock_get_projects, mock_slack_client, mock_slack_command, mock_project
+):
+    mock_get_projects.return_value = {"hits": [mock_project]}
+    settings.SLACK_COMMANDS_ENABLED = True
+
+    handler(ack=MagicMock(), command=mock_slack_command, client=mock_slack_client)
+
+    blocks = mock_slack_client.chat_postMessage.call_args[1]["blocks"]
+    assert any("Here are top 10 most OWASP projects" in str(block) for block in blocks)
```
Comment (arkid15r) [inline backend/tests/slack/commands/projects_tests.py]: Could you fix the wording here:
"Here are top 10 most relevant OWASP projects"

File: backend/tests/slack/events/user_joined_channel/contribute_tests.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        "apps.slack.events.user_joined_channel.contribute", fromlist=["contribute_handler"]
+    )
+    check_contribute_handler = getattr(
+        contribute_module,
+        "check_contribute_handler",
+        lambda x: x.get("channel") == OWASP_CONTRIBUTE_CHANNEL_ID,
+    )
+
+    assert check_contribute_handler({"channel": OWASP_CONTRIBUTE_CHANNEL_ID}) is True
+    assert check_contribute_handler({"channel": "C999999"}) is False
```
Comment (arkid15r) [inline backend/tests/slack/events/user_joined_channel/contribute_tests.py]:
```suggestion
    assert check_contribute_handler({"channel": OWASP_CONTRIBUTE_CHANNEL_ID})
    assert not check_contribute_handler({"channel": "C999999"})
```

**Review by arkid15r — CHANGES_REQUESTED** [review]
Let's make tests part of a class instead of just functions

Also let's use asserts w/o `is True` and `is False` -- just `assert` and `assert not`

File: backend/tests/slack/commands/projects_tests.py (hunk 2 above)
Comment (Yashgupta9330) [reply backend/tests/slack/commands/projects_tests.py]: but in projects.py Here are top 10 most OWASP projects is used

**Pull request merged**